js/packages/core/lib/dynamics.ts

```diff
@@ -44,22 +44,70 @@ export function compress(
   xn: ElemNode,
 ): NodeRepr_t;
 
-export function compress(
+export function compress(a_, b_, c_, d_, e_, f_, g_?) {
+  let children = (typeof a_ === "number" || isNode(a_))
+    ? [a_, b_, c_, d_, e_, f_]
+    : [b_, c_, d_, e_, f_, g_];
+
+  const [atkMs, relMs, threshold, ratio, sidechain, xn] = children;
+  const env = el.env(
+    el.tau2pole(el.mul(0.001, atkMs)),
+    el.tau2pole(el.mul(0.001, relMs)),
+    sidechain,
+  );
+
+  const envDecibels = el.gain2db(env);
+
+  // Calculate gain multiplier from the ratio (1 - 1/ratio)
+  const adjustedRatio = el.sub(1, el.div(1, ratio));
+
+  // Calculate gain reduction in dB
+  const gain = el.mul(adjustedRatio, el.sub(threshold, envDecibels)) // adjustedRatio * (threshold - envDecibels);
+
+  // Ensuring gain is not positive
+  const cleanGain = el.min(0, gain);
+
+  // Convert the gain reduction in dB to a gain factor
+  const compressedGain = el.db2gain(cleanGain);
+
+  return el.mul(xn, compressedGain);
+}
+
+
+/* A simple softknee compressor with parameterized attack and release times,
+ * threshold, compression ratio and knee width. 
+ *
+ * Functions like regular el.compress when kneeWidth is 0.
+ *
+ * Users may drive the compressor with an optional sidechain signal, or send the
+ * same input both as the input to be compressed and as the sidechain signal itself
+ * for standard compressor behavior.
+ *
+ * @param {Node | number} atkMs – attack time in milliseconds
+ * @param {Node | number} relMs – release time in millseconds
+ * @param {Node | number} threshold – decibel value above which the comp kicks in
+ * @param {Node | number} ratio – ratio by which we squash signal above the threshold
+ * @param {Node | number} kneeWidth – width of the knee in decibels, 0 for hard knee
+ * @param {Node} sidechain – sidechain signal to drive the compressor
+ * @param {Node} xn – input signal to filter
+ */
+export function skcompress(
```

Comment:
Let's do the same thing here, remove that `props` argument here in the type signature, and then below let's do

```
export function skcompress(atkMs, relMs, threshold, ratio, kneeWidth, sidechain, xn)
```

and skip the whole optional arg stuff